File: practice/algorithms/warmup/time_conversion/TimeConversion.scala
object TimeConversion extends App {
  val line = io.Source.stdin.bufferedReader().readLine()

  val hours = line.substring(0, line indexOf ':').toInt
  val minutes = line.substring((line indexOf ':') + 1, line lastIndexOf ':').toInt
  val seconds = line.substring((line lastIndexOf ':') + 1, (line lastIndexOf ':') + 3).toInt

  val newHours = if (hours == 12) 0 else hours
  val shift = if(line.toLowerCase contains "pm") 12 else 0
  println("%02d:%02d:%02d".format((newHours + shift), minutes, seconds))
}

File: practice/algorithms/warmup/time_conversion/TimeConversion.java
import java.util.Scanner;

public class TimeConversion {
    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        String input = stdin.nextLine();
        int hours = Integer.parseInt(input.substring(0, input.indexOf(':')));
        int minutes = Integer.parseInt(input.substring(input.indexOf(':') + 1, input.lastIndexOf(':')));
        int seconds = Integer.parseInt(input.substring(input.lastIndexOf(':') + 1, input.lastIndexOf(':') + 3));
        int newHours;
        if(hours == 12) {
            newHours = 0;
        } else {
            newHours = hours;
        }
        int shift;
        if(input.toLowerCase().contains("pm")) {
            shift = 12;
        } else {
            shift = 0;
        }
        System.out.println(String.format("%02d:%02d:%02d", (newHours + shift), minutes, seconds));
    }
}
